### DISCUSSION TITLE
Convert to issue does not use a template

### DISCUSSION THREAD

**Discussion by blairio** [community/community#14010] [Projects and Issues] [answered]
If I make a draft on a project in projects beta, then press convert to issue and pick the repo, it creates an issue but bypasses the issue template chooser and then you get an issue that is missing the template and you have to figure out how to combine them.  I suggest, that when there templates in a repo, that converting an issue allows the selection of a template and then the new issue becomes body of draft plus template.

**Comment by mffap** [comment]
Yes. Was looking for this. Thanks for raising this feedback. I would love to be able to convert based on a issue template.

**Comment by kevinwhoffman** [comment]
Currently the side panel loses all convenience when I have to jump to the New Issue screen to copy/paste a template.

Ideally when the issue is converted from a draft and a repo is selected, that repository's issue templates are displayed for selection.

**Comment by joonalaine** [comment]
This should also happen when converting from a list of items using the 'Convert to issue' feature.

**Comment by Techboidesign** [comment]
I also have the same question here! It would **totally make sense** for you to be able to **select a template**

**Comment by mazlan** [comment]
+1 to this. Would be super useful to create issues from templates directly from the Project board instead of draft/blank issue.

**Comment by plamber** [comment]
+1. This would improve the development flow considerably.

**Comment by justinbarry** [comment]
I also would love this.

**Comment by AnnaKahnRaspPi** [comment]
+1. I end up doing a really hinky workaround where I have a list of drafts in one screen and then I write the issue on another screen, so this would shave a significant chunk of time off my issue writing.

**Comment by YaroShkvorets** [comment]
Must have feature and should be fairly easy to implement
There are at least 5 other discussions asking for that feature going back to 2021.

**Comment by lucavadnais** [comment]
Implementing this feature would make any workflow way more productive.

**Comment by simisimis** [comment]
+1. all the beautiful things that we could do if only this was implemented 🙊

**Comment by iustin94** [comment]
Surprised this is not implemented yet...

**Comment by cathagg** [comment]
+1 - This feature would improve the workflow immensely.

**Comment by fkim7** [comment]
+1- Please consider this otherwise we have manually select projects, assignees, sprints, etc.

**Comment by spurohit42** [comment]
+1 this would be really helpful for so many reasons others have highlighted

**Comment by guidanoli** [comment]
+1 This would be very helpful for the productivity of devs

**Comment by ennovatenow** [comment]
+1 We are using the Issue template to define a strict criteria for issue logging. This feature gap negates all that effort.

**Comment by chhs2131** [comment]
Hope this becomes possible soon. 😅

**Comment by AnnaKahnRaspPi** [comment, marked as answer]
folks! my team worked out how to do this! 

there's a slash command - once you've converted the issue, if you click edit and type `/template` you can select one of your templates.

**Reply by stefan505** [reply]
@AnnaKahnRaspPi , I just tried this and it doesn't detect any of our existing issue templates:

![Screenshot 2023-06-12 at 10 20 36](https://github.com/community/community/assets/10812446/ffecdc52-774d-4a18-9d69-2c5975400c45)

Any suggestions?

**Reply by simisimis** [reply]
@stefan505 I have a template defined in a yml file, that is not seen by `templates` macro.
I followed documentation that led me to creating template from project settings, that created a file `.github/ISSUE_TEMPLATE/new-task.md`.
with content:
```
---
name: "New task"
about: Describe this issue template's purpose here.
title: ''
labels: ''
assignees: ''

---
some template text
```

This md file now does appear when calling templates macro.
Maybe my yml template has fields that template macro does not support or maybe it being in beta just supports md format.

**Reply by AnnaKahnRaspPi** [reply]
@simisimis I'm glad you knew the answer, because I didn't! The templates my engineers had set up just already worked.

**Reply by stefan505** [reply]
Thanks @simisimis and @AnnaKahnRaspPi , our templates are `yml`-based, will investigate converting to `md`. Take note @BOTLANNER

**Reply by runerei** [reply]
This is a nice solution to insert the template text, however it doesn't set labels and other fields.

**Reply by clintfred** [reply]
Aren't the yml templates the recommended way to create issue templates? Even if those worked, it seems like this is a nice workaround, but not really a "solution" per se. Is this being tracked anywhere else for full integration?

**Reply by johnfrazieratms** [reply]
Nice find @AnnaKahnRaspPi but as @simisimis pointed out, it doesn't work with `yml` templates and only `md` templates.  @rileybroughten is this a bug or a "feature"?

**Reply by joaomas** [reply]
@AnnaKahnRaspPi  could you show a step by step, because it's not working for me, even though I already have a template for issues but the drafts don't see my templates files just issues in dialog menu in project.

Thanks.

**Reply by simisimis** [reply]
@joaomas you can try searching github for inspiration. Check how templates are defined. Which exact directory they're placed in.
https://github.com/search?q=path%3A.github%2FISSUE_TEMPLATE%2F*.md&type=code
once you merge your `md` templates in your repo, then whenever you type in description `/template` macro you get to choose from a list of your `md` templates.

It also works with gh cli
`gh issue create -T "template name"`

It's a pity yaml based templates don't show up but this thread is already a couple of years old and in my case I am happy I chose to use more primitive md templates and was able to use them both in cli and in any existing or new issues, while `yaml` based you can only use when pressing a "New Issue" button.

**Reply by joaomas** [reply]
Hi @simisimis , thanks, but I don't have problem to create issue with Template, the question is: "draft -> convert to issue with template".

**Reply by simisimis** [reply]
Other than doing an extra step: 
draft > convert to issue > edit description and call template macro from there

I don't know if there's anything else you could do

**Comment by zsloane** [comment]
This also happens when you are converting a task in a tasklist (see: #72731)

As others pointed out in the accepted answer, this workaround is only good for getting the issue body text (and doesn't work for ```yml```-based templates?) and won't help with adding to project, setting labels etc etc
